src/util.h

```diff
@@ -83,6 +83,218 @@ extern CTranslationInterface translationInterface;
 extern const char *const BITCOIN_CONF_FILENAME;
 extern const char *const BITCOIN_PID_FILENAME;
 
+/** Send a string to the log output */
+int LogPrintStr(const std::string &str);
+
+// Logging API:
+// Use the two macros
+// LOG(ctgr,...)
+// LOGA(...)
+// located further down.
+// (Do not use the Logging functions directly)
+namespace Logging
+{
+extern uint64_t categoriesEnabled;
+
+/*
+To add a new log category:
+1) Create a unique 1 bit category mask. (Easiest is to 2* the last enum entry.)
+   Put it at the end of enum below.
+2) Add an category/string pair to LOGLABELMAP macro below.
+*/
+
+// Log Categories:
+// 64 Bits: (Define unique bits, not 'normal' numbers)
+enum
+{
+    NONE = 0x0, // No logging
+    ALL = 0xFFFFFFFFFFFFFFFFUL, // Log everything
+
+    // LOG Categories:
+    THN = 0x1,
+    MEP = 0x2,
+    CDB = 0x4,
+    TOR = 0x8,
+
+    NET = 0x10,
+    ADR = 0x20,
+    LIB = 0x40,
+    HTP = 0x80,
+
+    RPC = 0x100,
+    PRT = 0x200,
+    BNC = 0x400,
+    PRN = 0x800,
+
+    RDX = 0x1000,
+    MPR = 0x2000,
+    BLK = 0x4000,
+    EVC = 0x8000,
+
+    PRL = 0x10000,
+    RND = 0x20000,
+    REQ = 0x40000,
+    BLM = 0x80000,
+
+    EST = 0x100000,
+    LCK = 0x200000,
+    PRX = 0x400000,
+    DBS = 0x800000,
+    SLC = 0x1000000,
+
+};
+
+// Add corresponding upper case string for the category:
+#define LOGLABELMAP                                                                                           \
+    {                                                                                                         \
+        {NONE, "NONE"}, {ALL, "ALL"}, {THN, "THN"}, {MEP, "MEP"}, {CDB, "CDB"}, {TOR, "TOR"}, {NET, "NET"},   \
+            {ADR, "ADR"}, {LIB, "LIB"}, {HTP, "HTP"}, {RPC, "RPC"}, {PRT, "PRT"}, {BNC, "BNC"}, {PRN, "PRN"}, \
+            {RDX, "RDX"}, {MPR, "MPR"}, {BLK, "BLK"}, {EVC, "EVC"}, {PRL, "PRL"}, {RND, "RND"}, {REQ, "REQ"}, \
+            {BLM, "BLM"}, {LCK, "LCK"}, {PRX, "PRX"}, {DBS, "DBS"}, {SLC, "SLC"},                             \
+        {                                                                                                     \
+            EST, "EST"                                                                                        \
```

Comment:
why this new lines for "EST"

Reply:
auto-formatter seems to be doing this.  Everything else about this PR looks good to me.  Let's get this in, and the figure out what's going on with the formatter separately.  We need to start converting the logs to this new format anyway so there are more PRs to be done in this area.

Checking source formatting style...
(cd ../src; ../contrib/devtools/clang-format.py check clang-format-3.8 `cat ./.formatted-files`)
/fast/bitcoin/budev/contrib/devtools/trailing-comment.py
Detected clang-format version 3.8.0
--- util.h
+++ util.h_formatted
@@ -151,7 +151,9 @@
             {ADR, "ADR"}, {LIB, "LIB"}, {HTP, "HTP"}, {RPC, "RPC"}, {PRT, "PRT"}, {BNC, "BNC"}, {PRN, "PRN"}, \
             {RDX, "RDX"}, {MPR, "MPR"}, {BLK, "BLK"}, {EVC, "EVC"}, {PRL, "PRL"}, {RND, "RND"}, {REQ, "REQ"}, \
             {BLM, "BLM"}, {LCK, "LCK"}, {PRX, "PRX"}, {DBS, "DBS"}, {SLC, "SLC"},                             \
-        { EST, "EST" }                                                                                        \
+        {                                                                                                     \
+            EST, "EST"                                                                                        \
+        }                                                                                                     \
     }
 
Improper formatting found in: util.h
To properly format these files run: ../contrib/devtools/clang-format.py format clang-format-3.8 util.h